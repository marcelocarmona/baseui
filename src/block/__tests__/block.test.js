/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/

// @flow

import * as React from 'react';
import {mount} from 'enzyme';
import Block from '../block';

describe('Block', () => {
  it('renders correctly', () => {
    expect(mount(<Block color="primary200">test</Block>)).toMatchSnapshot();
  });

  it('applies other props if provided', () => {
    const onClickMock = jest.fn();
    const example = mount(<Block onClick={onClickMock}>test</Block>);
    example.simulate('click');

    expect(onClickMock).toHaveBeenCalled();
  });
});
