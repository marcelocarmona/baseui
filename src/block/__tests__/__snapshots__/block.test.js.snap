// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block renders correctly 1`] = `
<Block
  alignContent="stretch"
  alignItems="stretch"
  alignSelf="stretch"
  as="div"
  color="primary200"
  display="block"
  flex="shrink"
  flexWrap={false}
  justifyContent="start"
  overflow="visible"
  position="static"
>
  <MockStyledComponent
    $alignContent="stretch"
    $alignItems="stretch"
    $alignSelf="stretch"
    $as="div"
    $color="primary200"
    $display="block"
    $flex="shrink"
    $flexWrap={false}
    $justifyContent="start"
    $overflow="visible"
    $position="static"
  >
    <div
      styled-component="true"
    >
      test
    </div>
  </MockStyledComponent>
</Block>
`;
